app/models/image_variant.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+# Helper class to translate between variant names and dimensions.
+class ImageVariant
+  VARIANTS = {
+    tiny: [25, 25],
+    small: [50, 50],
+    large: [125, 125],
+    album: [200, 200],
+    original: [800, 800],
+    greenfield: [1500, 1500],
+    hq: [3000, 3000]
+  }.freeze
+
+  # Fit a variant inside these dimensions (width x height) to generate the variant.
+  def self.resize_to_fit(variant_name)
+    VARIANTS[variant_name.to_sym]
+  end
+
+  # Raises ArgumentError with explanation when the variant name does not exist.
+  def self.verify(variant_name)
```

Comment:
What's the reasoning to have the error handling code here? It seems "fluffy" but maybe there's something I'm missing. For example, if we remove 25x25 variants, we can insure no place in the code is asking for variants of that size. If for some reason the code asks for an unlisted size, what's the benefit of the method blowing up in a "fancy" way vs. "normal"?

Reply:
I guess I added this instinctively. It's mostly to help front-end focussed developers figure out what's going on when they have a typo in the variants. Otherwise they might get an exception from deep in the stack somewhere about something being `nil`. I can keep the code lean and remove this if you want.

Reply:
One of the issue with Paperclip is that it will happily generate a URL to a non-existent 'style'.

```ruby
Pic.first.url(variant: :mew) #=>
  "https://d15x8imun0vkea.cloudfront.net/pics/1/construction_mew.jpg"
```

Reply:
Ah gotcha! Usually I'd vote to keep things lean, but actually we've had a history of broken images and so on, so I'm happy for things to go a couple clicks more explicit. Thanks!